python_modules/libraries/dagster-components/dagster_components/lib/custom_component.py

```diff
@@ -0,0 +1,87 @@
+import os
+from abc import abstractmethod
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, Mapping, Optional
+
+import click
+from dagster._utils import snakecase
+from pydantic import BaseModel
+
+from dagster_components.core.component import (
+    Component,
+    ComponentGenerateRequest,
+    ComponentLoadContext,
+    component,
+)
+from dagster_components.generate import generate_custom_component_yaml
+
+if TYPE_CHECKING:
+    from dagster._core.definitions.definitions_class import Definitions
+
+CUSTOM_COMPONENT_TEMPLATE = '''from dagster import Definitions
```

Comment:
I think we would be better off putting this as real Python code in a function like this:

```
def custom_component_template():
    ...
```

Then you can get the source out as a string with:

```
source = textwrap.dedent(inspect.getsource(custom_component_template).split("\n", 1)[1])
```

This allows us to keep the code linted and type-checked. You could replace `{class_name}` with a legal python identifier for use as a placeholder and then just `.replace(PLACEHOLDER, class_name}` it.